server/app.ts

```diff
@@ -54,6 +54,10 @@ export async function main() {
 	await buildClients();
 	buildRateLimiter();
 
+	if (process.argv.includes("--validate")) {
+		validateConfig();
+	}
```

Comment:
This needs to happen before the config gets used to do anything. For example, if you set `redis.host` to a valid host, but the host is not available, this will hang on `buildClients()` retrying forever until it connects to redis. Move it to right under the `setLogLevel` call